src/Spago/Build.hs

```diff
@@ -43,6 +43,14 @@ data NoBuild = NoBuild | DoBuild
 -- | Flag to skip the automatic installation of libraries on build
 data NoInstall = NoInstall | DoInstall
 
+-- | Only build deps and ignore input paths
+data DepsOnly = DepsOnly | WithAll
+
+applyDepsOnly :: DepsOnly -> [Purs.SourcePath] -> [Purs.SourcePath]
+applyDepsOnly depsOnly sources = case depsOnly of
```

Comment:
Since in every place we call `Packages.getGlobs` we also attach the project sources, lets:
- take `depsOnly` and the `Config` as parameters in `getGlobs`
- do this check there instead of doing this check and concatenating the project paths all over the place